homeassistant/components/lutron/light.py

```diff
@@ -25,14 +31,38 @@ async def async_setup_entry(
     Adds dimmers from the Main Repeater associated with the config_entry as
     light entities.
     """
+    ent_reg = er.async_get(hass)
     entry_data: LutronData = hass.data[DOMAIN][config_entry.entry_id]
-    async_add_entities(
-        [
-            LutronLight(area_name, device, entry_data.client)
-            for area_name, device in entry_data.lights
-        ],
-        True,
-    )
+
+    for area_name, device in entry_data.lights:
+        if device.type == "CEILING_FAN_TYPE":
+            # If this is a fan check to see if this entity already exists. If not, do not create a new one.
+            entity_id = f"light.{slugify(f'{area_name} {device.name}')}"
+            if ent_reg.async_is_registered(entity_id):
+                entity_entry = ent_reg.async_get(entity_id)
+                assert entity_entry
+                if entity_entry.disabled:
+                    # If the entity exists and is disabled then we want to remove the entity so that the user is using the new fan entity instead.
+                    ent_reg.async_remove(entity_id)
+                else:
+                    async_add_entities(
+                        [LutronLight(area_name, device, entry_data.client)],
+                        True,
+                    )
+                    async_create_issue(
```

Comment:
We shouldn't create a repair issue just because the entity is enabled. We don't know that the user is using it.

There's a helper to check if an entity is referenced in automations and scripts. Use that to create a repair issue.

Reply:
Example:
https://github.com/home-assistant/core/blob/4a34cd25b2130afe254ae0500b552d1c65f63e9f/homeassistant/components/harmony/switch.py#L96-L99

Reply:
the idea here is that we want the user to disable all fan entities previously created as a light.  Then on the next restart of the integration they will be removed and the user will be migrated.  So if it is enabled, it is a problem for us moving forward.  Once disabled, we know the deprecated code is no longer needed by them.

Reply:
I tested your example and it leaves a big gap.  In a lot of cases the user will have scenes setup in the lutron device that home assistant doesn't control other than to toggle the scene.  The user might often run all scenes out of the Lutron stack entirely and then just use the entity in HA for manual control or state.  In combination with the above, I think we should raise the issue for any enabled fan and not just ones included in automations and scenes.